### PR TITLE
mgr/dashboard: Show smartctl summary in OSD tab

### PR DESCRIPTION
**Pull request by p-se**
It extends the "Device health" tab on the OSD page and shows a the smartctl summary of the device above the "Device information" and "S.M.A.R.T" tab of the selected device.

It contains the refactoring of `InfoPanelComponent`, `WarningPanelComponent` and `ErrorPanelComponent` and combines those into the `AlertPanelComponent`. A separate commit adds the success type.

Personally, I found it disturbing to have such a big colored panel visible all the time above the tabs, but I like that it uses colors to show if the overall check has passed or failed. That's why I added the size 'slim' to the AlertComponent. This is what it looked like before and how it looks now. Please let me know what you think and if you disagree with this additional option.

Fixes: https://tracker.ceph.com/issues/41774

[This patch](https://gist.githubusercontent.com/p-se/fa58d11a95d0192130627b002c53ffac/raw/c2007d5fbb8330d1a484908ac80d86e955c2e185/mgr_dashboard__enable_the_user_to_download_copy_a_smartctl_report_of_a_device__Fixes__http1.patch) can be applied to the backend to **fix the response of the backend** in case it doesn't work for any reason. For possible reasons and other options, see the [description of the PR that introduced the SMART functionality](https://github.com/ceph/ceph/pull/30324#issue-316393144).

***

Before

***

**Slim with icon (that has become it, thanks for your feedback!)**

***

Slim without icon

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by callithea** [comment]
I'd vote for the "Slim with icon" version.

**Comment by tspmelo** [comment]
For this situation I prefer the "slim with icon". 
Would it be possible to pass an attr for the component to change from 1 to the other display?
I think the bigger version will look better when its displayed in a black page...

**Comment by p-se** [comment]
> Would it be possible to pass an attr for the component to change from 1 to the other display?
> I think the bigger version will look better when its displayed in a black page...

The component has an attribute `size` already, that currently accepts two values: `normal` and `slim`. `normal` is the default and doesn't need to be specified. `slim` is the one I used here. It has also an attribute for the icons. Nothing of the screenshots was made up, it's possible to let the alerts look like the screenshots by simply passing arguments to the `AlertPanelComponent`.

**Comment by p-se** [comment]
jenkins test make check

**Comment by p-se** [comment]
jenkins test make check

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-smart-list/osd-smart-list.component.html [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
         <ng-container *ngIf="device.value.error; else noError">
-          <cd-warning-panel>{{ device.value.userMessage }}</cd-warning-panel>
+          <cd-alert-panel type="warning">{{ device.value.userMessage }}</cd-alert-panel>
         </ng-container>
         <ng-template #noError>
+          <ng-container *ngIf="device.value.smart.data.self_test.status.passed; else selfTestFailed">
+            <cd-alert-panel
+              size="slim"
+              type="info"
+              title="SMART overall-health self-assessment test result">
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-smart-list/osd-smart-list.component.html]: Missing translation.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-smart-list/osd-smart-list.component.html [hunk 2]
```diff
@@ -1,14 +1,30 @@
...
+              {{ device.value.smart.data.self_test.status.string }}
+            </cd-alert-panel>
+          </ng-container>
+          <ng-template #selfTestFailed>
+            <cd-alert-panel
+              size="slim"
+              type="warning"
+              title="SMART overall-health self-assessment test result">
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-smart-list/osd-smart-list.component.html]: Missing translation.

**Comment by votdev** [comment]
I'm struggling a little bit how this PR is named. IMO the adaption for the new alert panel takes more attention in this PR than the small changes for the OSD page.

**Comment by p-se** [comment]
> I'm struggling a little bit how this PR is named. IMO the adaption for the new alert panel takes more attention in this PR than the small changes for the OSD page.

That's why it's a separate commit and properly described in that commit, but as the union of three different kinds of panels into a single component is just a refactoring, I haven't thought it deserves it own PR, even less should the PR be named after this refactoring. It's about the addition of alert panels into the smart component. Also, I don't think this union is very questionable (and you haven't requested any changes on that refactoring, either), so I kept it here. And last but not least, the newly compound component is also used by this implementation. If I separated them, I'd have, once more, a PR that depends on another (either way). Don't you agree?

**Comment by p-se** [comment]
jenkins test make check arm64

**Pull request merged**